lib/pages/new_private_chat/fetch_contacts_controller.dart

```diff
@@ -0,0 +1,25 @@
+
+import 'dart:async';
+
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/app_state/contact/get_contacts_success.dart';
+import 'package:fluffychat/domain/usecase/fetch_contacts_interactor.dart';
+
+class FetchContactsController {
+  late final FetchContactsInteractor _fetchContactsInteractor = getIt.get<FetchContactsInteractor>();
+  final StreamController<Either<Failure, GetContactsSuccess>> streamController = StreamController();
+
+  void fetchCurrentTomContacts() {
+    _fetchContactsInteractor
+      .execute()
+      .listen((event) {
+        streamController.add(event);
+      });
+  }
+
+  void dispose() {
+    streamController.close();
```

Comment:
perhaps you should end the stream interactor aswell here

Reply:
_fetchContactsInteractor is a Interactor, so i can't close any stream here. _fetchContactsInteractor will be disposed when i call unbind in my ContactDI. so no worry